evals/solvers/providers/fixie/local_gpu_solver.py

```diff
@@ -255,7 +256,9 @@ def batch_requests(self):
                     item = self._batch_queue.get(block=False)
                     work_items.append(item)
                 except queue.Empty:
-                    break
+                    if self.process_pool_executor._call_queue.empty():
+                        break
+                    time.sleep(0.01)
```

Comment:
I don't love the dependency on `ProcessPoolExecutor`'s internals, especially since it seems tractable to track this on our side. (My inclination would still be to manage the process pool queue depth more explicitly with a semaphore but I suspect in steady state it doesn't matter much.)

But the logic itself is sensible enough: stop batching when either the batch is full or the process pool is ready (or nearly ready) for work.

Reply:
Modified to be handled internally.